pkg/scheduler/context.go

```diff
@@ -918,3 +921,11 @@ func (cc *ClusterContext) SetRMInfo(rmID string, rmBuildInformation map[string]s
 		RMBuildInformation: buildInfo,
 	}
 }
+
+func (cc *ClusterContext) GetHealthCheckCache() *dao.SchedulerHealthDAOInfo {
+	return cc.healthCheckCache
+}
+
+func (cc *ClusterContext) SetHealthCheckCache(c *dao.SchedulerHealthDAOInfo) {
+	cc.healthCheckCache = c
+}
```

Comment:
These methods are invoked from different goroutines. Add the necessary `RLock()` and `RUnlock()` calls.